libethashseal/EthashCPUMiner.cpp

```diff
@@ -62,7 +62,7 @@ void EthashCPUMiner::workLoop()
 	uint64_t tryNonce = s_eng();
 	ethash_return_value ethashReturn;
 
-	WorkPackage w = work();
+	WorkPackage const& w = work();
```

Comment:
We need a copy here. The `work()` has a lock inside, so accessing the work package by pointer (aka reference) will overpass it. The best fix is to return from `work()` by value.